tools/certgen/certgen.go

```diff
@@ -46,14 +45,25 @@ func main() {
 }
 
 func createCertificateViaFlags() error {
-	return certgenInternal.CreateCertificateAndPrivateKeyFiles(&certgenInternal.Params{
-		CertificateAuthority:      *caFlag,
-		RootPublicCertificatePath: *rootPublicCertificateFlag,
-		RootPrivateKeyPath:        *rootPrivateKeyFlag,
-		PublicCertificatePath:     *publicCertificateFlag,
-		PrivateKeyPath:            *privateKeyFlag,
-		ValidityDuration:          *validityDurationFlag,
-		Hostnames:                 *hostnamesFlag,
-		RSAKeyLength:              *rsaKeyLengthFlag,
-	})
+	params := &certgenInternal.Params{
+		CertificateAuthority: *caFlag,
+		ValidityDuration:     *validityDurationFlag,
+		Hostnames:            strings.Split(*hostnamesFlag, ","),
+		RSAKeyLength:         *rsaKeyLengthFlag,
+	}
+
+	if len(*rootPublicCertificateFlag) > 0 {
```

Comment:
This still has a weird affordance where using the private root flag (but not the public) will silently not do what you want, and using public root flag (but missing the private) will cause a file not found error.  When an API is misused, it should have clear reaction about the misuse.

My suggested fix which handles this at both the command level and the library level is:
Split this if statement into two, one for public, one for private.
At the beginning of CreateCertificateAndPrivateKeyFiles, check:
`
if (params.RootPublicCertificateData == nil) != (params.RootPrivateKeyData) {
  return errors.New("Root Certificate Public and Private must both be set, or neither be set")
}
`